src/horizontalaligner.cpp

```diff
@@ -33,6 +34,124 @@
 
 namespace vrv {
 
+//----------------------------------------------------------------------------
+// Fraction
+//----------------------------------------------------------------------------
+
+Fraction::Fraction(int num, int denom) : m_numerator(num), m_denominator(denom)
+{
+    if (denom == 0) {
+        LogDebug("Denominator cannot be zero.");
+        denom = 1;
+    }
+    Reduce();
+}
+
+Fraction::Fraction(data_DURATION duration)
+{
+    duration = vrv::DurationMin(duration, DURATION_1024);
+    duration = vrv::DurationMax(duration, DURATION_NONE);
+    int den = pow(2, (duration + 1));
```

Comment:
`DURATION_NONE` is defined as -2 which gives pow(2, -1) on the rhs which is 0.5. Now casting to int (which is truncating) gives `den = 0`. I think this edge case still needs clarification.

Reply:
You are right. I will change `DURATION_NONE` to `DURATION_maxima`.